txnkv/transaction/txn.go

```diff
@@ -742,6 +742,8 @@ func (txn *KVTxn) Rollback() error {
 		}
 	}
 	if txn.IsPipelined() && txn.committer != nil {
+		// wait all flush to finish, this avoids data race.
+		txn.GetMemBuffer().FlushWait()
```

Comment:
Is there any convenient way to avoid this wait? I suppose Rollback operations had better be immediate. This is not a big problem any way. Don't be blocked by it.

Reply:
It's not allowed that rollback and flush running concurrently.
Added a cancel function to avoid wait too long.

Reply:
> It's not allowed that rollback and flush running concurrently.

When the rollback has put the rollback record at the primary lock, it shoule be ok to be executed concurrently, the stale prewrite lock would be cleaned up by the rollback requests or they would encounter write conflict seeing the rollback record.

Reply:
@cfzjywxk It's ok for the distribution-txn stores, but race may happens inside client-go.

`txn.diskFullOpt = kvrpcpb.DiskFullOpt_NotAllowedOnFull` is invoked inside `txn.close()`.

https://github.com/tikv/client-go/blob/b81a5ee7c43f5e34aced4a9aa72cb3701acd2217/txnkv/transaction/txn.go#L747

Meanwhile, `txn.diskFullOpt` may be read here.

https://github.com/tikv/client-go/blob/b81a5ee7c43f5e34aced4a9aa72cb3701acd2217/txnkv/transaction/pipelined_flush.go#L99